### PR TITLE
Using Batch Norm & Using numeric check to cope with NaN without interrupt training

### PR DESCRIPTION
**Pull request by yhyu13**
Using batch norm layer to achieve normalized inputs. Change batch size to 50. Using reduce_mean to judge loss independently from batch size.

### CODE REVIEW

**Comment by yhyu13** [comment]
Using numeric check is effective
discard_nan

You can now review real time tensorboard (because it used to be confused by models checkpoints)
realtime
